File: zap-scripts/auth-header.js
// ZAP HTTP Sender Script to inject custom Authorization + accept headers
// This runs before every request ZAP makes.

function sendingRequest(msg, initiator, helper) {
    var requestHeader = msg.getRequestHeader();

    // Inject your static Authorization value
    requestHeader.setHeader("Authorization", "YW5OdGFYUm86WkdWdGJ6RXlNelE9Oj8DP1IWYC0=");

    // Ensure JSON Accept header
    requestHeader.setHeader("accept", "application/json");

    msg.setRequestHeader(requestHeader);
}

function responseReceived(msg, initiator, helper) {
    // no-op
}

File: zap-scripts/auth-header.py
# Inject Authorization header into every request made by ZAP

def sendingRequest(msg, initiator, helper):
    header = msg.getRequestHeader()
    header.setHeader("Authorization", "YW5OdGFYUm86WkdWdGJ6RXlNelE9Oj8DP1IWYC0=")
    header.setHeader("accept", "application/json")

def responseReceived(msg, initiator, helper):
    pass
